docs/query-metadata-style-guide.md

```diff
@@ -0,0 +1,152 @@
+
+*   C and C++: `cpp`
+*   C#: `cs`
+*   COBOL: `cobol`
+*   Java: `java`
+*   JavaScript and TypeScript: `js`
+*   Python: `py`
+
+The `@id` should consist of a short noun phrase that identifies the issue that the query highlights. For example:
+
+
+
+*   `@id cs/command-line-injection`
+*   `@id java/string-concatenation-in-loop`
+
+Note, `@id` properties should be consistent for queries that highlight the same issue for different languages. For example, the following queries identify format strings that contain unsanitized input in Java and C++ code respectively:
+
+
+
+*   `@id java/tainted-format-string`
+*   `@id cpp/tainted-format-string`
+
+
+### Query type `@kind`
+
+`@kind` is a required property that defines the type of query. The main query types are:
+
+
+
+*   alerts (`@kind problem`)
+*   alerts containing path information (`@kind path-problem`)
+
+These `@kind` properties support two further mandatory properties which are added by Semmle after the query has been tested, prior to deployment to LGTM. The following information is for reference:
+
+
+
+*   `@precision`–broadly indicates the proportion of query results that are true positives, while also considering their context and relevance:
+    *   `low `
+    *   `medium `
+    *   `high `
+    *   `very high`
+*   `@problem.severity`–defines the level of severity of the alert: 
+    *   `error`–an issue that is likely to cause incorrect program behavior, for example a crash or vulnerability.
+    *   `warning`–an issue that indicates a potential problem in the code, or makes the code fragile if another (unrelated) part of code is changed.
+    *   `recommendation`–an issue where the code behaves correctly, but it could be improved.
+
+The values of `@precision` and `@problem.severity` assigned to a query that is part of the standard set determine how the results are displayed by LGTM. See [About alerts](https://help.semmle.com/lgtm-enterprise/user/help/about-alerts.html) and [Alert interest](https://lgtm.com/help/lgtm/alert-interest) for further information. For information about using custom queries in LGTM on a 'per-project' basis, see [Writing custom queries to include in LGTM analysis](https://lgtm.com/help/lgtm/writing-custom-queries) and [About adding custom queries](https://help.semmle.com/lgtm-enterprise/admin/help/about-adding-custom-queries.html). 
+
+## Query tags `@tags`
+
+The `@tags` property is used to define categories that the query relates to. Each query should belong to one (or more, if necessary) of the following four top-level categories:
+
+*   `@tags correctness`–for queries that detect incorrect program behavior.
+*   `@tags maintainability`–for queries that detect patterns that make it hard for developers to make changes to the code.
+*   `@tags readability`–for queries that detect confusing patterns that make it hard for developers to read the code.
```

Comment:
In both `maintainability` and `readability` I would prefer the wording "..make it harder for developers.." to the current "..make it hard for developers..", as "harder" indicates that things are more difficult than they could otherwise be, which is probably in line with what we claim in the various queries, but claiming that things are "hard" is a slightly different judgment that I don't think we can claim.